### PR TITLE
Summarize long non-JSON function responses

### PR DESCRIPTION
**Pull request by Henry-Pulver**
If a non-JSON function response is >1500 tokens, we split it into 2000 token chunks and in parallel summarise each, then join the summaries of each chunk up.

**To test on staging:**
1. Go on the jimbob.rowland account
2. Ask Henry for the API key

https://github.com/Superflows-AI/superflows/assets/33871096/9d0a0f11-5369-496a-947b-0eaf4f623eab

### CODE REVIEW

File: lib/prompts/summarizeText.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  organization: Organization
+): ChatGPTMessage[] {
+  return [
+    {
+      role: "system",
+      content: `Your task is to summarize a document using bullet points and short simple sentences.
+
+For context of what to include in your summary, you are working for ${organization.name}. ${organization.description}
+
+Your response should be a maximum of 6 bullet points. Only use one, two or three bullet points if you can. Only write short simple sentences. Only write one sentence per bullet point. Be as succinct as possible.
```
Comment (neuromantic99) [inline lib/prompts/summarizeText.ts:22]: This is probably hard but the summary often doesn't contain the information that you'd actually want to see:

**original**:
image

**Summary**
image

I would want the summary to say e.g. 
"the net assests for the company is X" rather than "Balance sheet shows fixed assets, current assets, and net assets."

Comment (neuromantic99) [reply lib/prompts/summarizeText.ts:22]: maybe we need a line like: "be specific about the facts you include, include numerical information"

File: pages/api/v1/answers.ts [hunk 2]
```diff
@@ -562,101 +568,110 @@ async function Angela( // Good ol' Angela
...
+                role: "function",
+                name: command.name,
+                content:
+                  typeof out === "string" ? out : JSON.stringify(out, null, 2),
+              };
+
+              // If >1500 tokens, summarise the message
+              if (
+                typeof out === "string" &&
+                getTokenCount([outMessage]) > 1500
```
Comment (neuromantic99) [inline pages/api/v1/answers.ts:657]: this threshold seems quite high, is a 1000 word message not worth summarising? that's like 3 pages of a book

File: components/playgroundChatbot.tsx [hunk 3]
```diff
@@ -358,25 +360,19 @@ export default function PlaygroundChatbot(props: {
...
                 contentString = "No data returned";
-              } else if (
-                functionJsonResponse &&
-                typeof functionJsonResponse === "object"
-              ) {
-                contentString = chatItem.content;
               }
               return (
-                <DevChatItem
+                <UserChatItem
```
Comment (neuromantic99) [inline components/playgroundChatbot.tsx:373]: why the change from Dev to User?

File: package.json [hunk 4]
```diff
@@ -18,7 +18,7 @@
     "@supabase/auth-ui-react": "^0.4.2",
     "@supabase/auth-ui-shared": "^0.1.6",
     "@supabase/supabase-js": "^2.26.0",
-    "@superflows/chat-ui-react": "^1.1.6",
+    "@superflows/chat-ui-react": "^1.1.6-beta.1",
```
Comment (neuromantic99) [inline package.json:21]: what's the plan for release here? we have a beta on npmjs?

Comment (Henry-Pulver) [reply package.json:21]: Pushing a beta allows us to test in staging :smile: 

Like usual, `chat-ui` needs to be merged and published (to `1.1.7`) and this updated before merging

File: lib/prompts/summarizeText.ts (hunk 1 above)
Comment (Henry-Pulver) [reply lib/prompts/summarizeText.ts:22]: Surprised it came out with this! Never seen it do this

Have this line already:
https://github.com/Superflows-AI/superflows/blob/summarise-long-function-responses/lib/prompts/summarizeText.ts#L24

Added a `THIS IS VERY IMPORTANT. DO NOT FORGET THIS!`

File: pages/api/v1/answers.ts (hunk 2 above)
Comment (Henry-Pulver) [reply pages/api/v1/answers.ts:657]: Good point, will cut it to 500.

I expected the summarisation to be slower and worse than it is, so wanted it to have all the information unless a response was super long

File: components/playgroundChatbot.tsx (hunk 3 above)
Comment (Henry-Pulver) [reply components/playgroundChatbot.tsx:373]: This change is in `chat-ui`. The change enables developer mode to show a different viz to non-developer mode :smile:

File: lib/prompts/summarizeText.ts (hunk 1 above)
Comment (neuromantic99) [reply lib/prompts/summarizeText.ts:22]: oh yeah that worked nicely when i just tried it
